demos/common/cpp/models/src/hpe_model_associative_embedding.cpp

```diff
@@ -0,0 +1,207 @@
+
+#include <utils/common.hpp>
+#include <utils/ocv_common.hpp>
+#include <ngraph/ngraph.hpp>
+
+using namespace InferenceEngine;
+
+const cv::Vec3f HpeAssociativeEmbedding::meanPixel = cv::Vec3f::all(128);
+const float HpeAssociativeEmbedding::detectionThreshold = 0.1f;
+const float HpeAssociativeEmbedding::tagThreshold = 1.0f;
+const float HpeAssociativeEmbedding::delta = 0.0f;
+const bool HpeAssociativeEmbedding::useDetectionVal = true;
+const bool HpeAssociativeEmbedding::doAdjust = true;
+const bool HpeAssociativeEmbedding::doRefine = true;
+const bool HpeAssociativeEmbedding::ignoreTooMuch = true;
+
+HpeAssociativeEmbedding::HpeAssociativeEmbedding(const std::string& modelFileName, double aspectRatio,
+    int targetSize, float confidenceThreshold) :
+    ModelBase(modelFileName),
+    aspectRatio(aspectRatio),
+    targetSize(targetSize),
+    confidenceThreshold(confidenceThreshold) {
+}
+
+void HpeAssociativeEmbedding::prepareInputsOutputs(CNNNetwork& cnnNetwork) {
+    // --------------------------- Configure input & output -------------------------------------------------
+    // --------------------------- Prepare input blobs ------------------------------------------------------
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    if (inputShapes.size() != 1)
+        throw std::runtime_error("Demo supports topologies only with 1 input");
+    inputsNames.push_back(inputShapes.begin()->first);
+    SizeVector& inSizeVector = inputShapes.begin()->second;
+    if (inSizeVector.size() != 4 || inSizeVector[0] != 1 || inSizeVector[1] != 3)
+        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
+    InputInfo& inputInfo = *cnnNetwork.getInputsInfo().begin()->second;
+    inputInfo.setPrecision(Precision::U8);
+    inputInfo.getInputData()->setLayout(Layout::NCHW);
+
+    // --------------------------- Prepare output blobs -----------------------------------------------------
+    const OutputsDataMap& outputInfo = cnnNetwork.getOutputsInfo();
+    if (outputInfo.size() != 3)
+        throw std::runtime_error("Demo supports topologies only with 3 outputs");
+    for (const auto& outputLayer: outputInfo) {
+        outputLayer.second->setPrecision(Precision::FP32);
+        outputsNames.push_back(outputLayer.first);
+        const SizeVector& outputLayerDims = outputLayer.second->getTensorDesc().getDims();
+        if (outputLayerDims.size() != 4 && outputLayerDims.size() != 5)
+                throw std::runtime_error("output layers are expected to be 4-dimensional or 5-dimensional");
+        if (outputLayerDims[0] != 1 || outputLayerDims[1] != 17)
+                throw std::runtime_error("output layers are expected to have 1 batch size and 17 channels");
+    }
+}
+
+void HpeAssociativeEmbedding::reshape(CNNNetwork& cnnNetwork) {
+    ICNNNetwork::InputShapes inputShapes = cnnNetwork.getInputShapes();
+    SizeVector& InputLayerDims = inputShapes.begin()->second;
```

Comment:
`InputLayerDims` name should start with lower case letter. Please update it for the both demos. Since the variable name is going to be changed any way please remove `Layer` word: layer is an operation, and here we care about data it takes, but not the operation itself.